File: 백준문제/C언어/1이될때까지.c
#include<stdio.h>

main()
{
	int n = 17;
	int k = 4;
	int cnt = 0;
	while (1)
	{
		printf("µø¿€¡ﬂ");
		if (n % k == 0)
			n /= k;
		else
			n--;
		cnt++;
		if (n == 1)
			break;
	}
	printf("%d", cnt);

}

File: 백준문제/Python/greedy/1이될때까지.py
n, k = map(int,input().split())
cnt = 0
while True:
    # n에서 k로 나누어떨어지는 수 중 가장 가까운 수
    # 예를들어 n=14 k=3이면 target = (14//3)*3 = 12
    # n이 target의 수가 되려면 2(n-target)번이 되어야 한다.
    target = (n//k) * k
    cnt += (n - target)
    n = target
    if n < k :
        break
    cnt += 1
    n //= k
cnt += (n-1)
print(cnt)